### ISSUE TITLE
Can't upload more after max hit

### ISSUE THREAD

**Issue by greggilbert** [thetutlage/vue-clip#31]
When setting `maxFiles` as part of options, if you upload the max number of files and then delete one, you cannot add more.

To reproduce:

1. Set `maxFiles: 3` in options
2. Add a bind to delete uploads, e.g.

```
    removeFile(file) {
      this.vendor.logo = null;
      var index = this.$refs.vc.files.indexOf(file);
      this.$refs.vc.files.splice(index, 1);
    },

          <template slot="clip-uploader-body" scope="props">
            <div v-for="file in props.files">
              <img v-bind:src="file.dataUrl" v-on:click="removeFile(file)">
              {{ file.name }} {{ file.status }}
            </div>
          </template>
```
3. Upload 3 images
4. Click on one of the images to fire `removeFile`
5. Try to upload another image and watch nothing happen, without any errors

**Comment by alexsasharegan** [comment]
This is probably related to a Dropzone problem. Vue-clip proxies almost all of its operations to Dropzone. I'll try take a look and see if the removeFile has internals of its own it just continues to proxy operations to Dropzone.

**Comment by alexsasharegan** [comment]
Try this: 

```js
removeFile(file) {
      this.vendor.logo = null;
      this.$refs.vc.removeFile(file);
    }
```

**Comment by greggilbert** [comment]
Hmm, that didn't work for me. It also left the uploaded file there with a status of `removed`.

**Comment by alexsasharegan** [comment]
Well :hankey: . I've been traveling, so haven't dove in just yet to figure out what the culprit is.

I remember maxFiles is managed entirely by Dropzone, so some method needs to be called to ensure DZ decrements it's file count. Vue-clip copies the DZ files to its own array (I believe that's to gain reactivity), so your original splice might still be necessary.

**Comment by alexsasharegan** [comment]
Yeah, the removeFile method proxies to Dropzone (but it's passing the copied file to DZ, so let's hope it removes the file and decrements). That method does not remove it from its own files array though. Try adding your splice call right after the line `this.$refs.vc.removeFile(file)`

**Comment by alexsasharegan** [comment]
Looks like it won't work because DZ is doing an equality check on the file object, and vue-clip creates a different file object, so the references are not equal. Checkout the DZ code that gets run on its files array: 

```js
  without = function(list, rejectedItem) {
    var item, j, len, results;
    results = [];
    for (j = 0, len = list.length; j < len; j++) {
      item = list[j];
      if (item !== rejectedItem) {
        results.push(item);
      }
    }
    return results;
  };
```

**Comment by alexsasharegan** [comment]
Unless you get this to work, I think this is inherently incompatible due to Dropzone's remove method. You could file an issue on their repo, but keep in mind it had moved to gitlab.

**Comment by alexsasharegan** [comment]
This is more or less what I would hope works:
```vue
<template>
<vue-clip ref="vc">
    <template slot="clip-uploader-body" scope="props">
        <div v-for="file in props.files">
            <img v-bind:src="file.dataUrl" v-on:click="removeFile(file)">
            {{ file.name }} {{ file.status }}
        </div>
    </template>
</vue-clip>
</template>

<script>
export default {
    removeFile(file) {
        // gets dropzone to decrement max files
        this.$refs.vc.removeFile(file);
        // gets vue clip to remove the file from it's array
        var index = this.$refs.vc.files.indexOf(file);
        this.$refs.vc.files.splice(index, 1);
    }
}
</script>

**Comment by aleksandrmelnyk** [comment]
Similar issue when you want to change maxFiles dynamically. Would be gr8 if vue-clip supports options update, at least like setMaxFiles() etc.

**Comment by aleksandrmelnyk** [comment]
I also used this:
`this.$refs.vc.uploader._uploader.options.maxFiles = 10`(updated)
though not sure if it will work on production build

**Comment by alexsasharegan** [comment]
Your workaround directly accesses the underlying dropzone instance, so it would work 👍 

For any others that try to replicate @aleksandrmelnyk's workaround, just remember that `this.$refs.vc` refers to his vue-clip ref _(`this.$refs[yourRefHere]`)_, the rest of that object path is generic.

\* If you want to give vue-transmit a try, you can just dynamically update your `maxFiles` to the vue-transmit uploader component: https://github.com/alexsasharegan/vue-transmit
